plugins/base/src/test/kotlin/renderers/html/HtmlRenderingOnlyTestBase.kt

```diff
@@ -48,13 +49,15 @@ abstract class HtmlRenderingOnlyTestBase : RenderingOnlyTestBase<Element>() {
         finalizeCoroutines = false
     )
 
-    override val context = MockContext(
+    override val context by lazy { context(configuration) }
+
+    fun context(testConfiguration: DokkaConfiguration): MockContext = MockContext(
```

Comment:
(if you stick to `HtmlRenderingOnlyTestBase` and it's not reverted)

Was there some sort of a problem with it and this refactoring helped?

Reply:
The problem is, while overall all of extensions are the same for all tests in file, configuration should be different per different tests - otherwise it's not possible to change some specific variables here.
F.e. in this case, I need to test output based on `configuration.sourceSets[i].sourceLinks`, and so `configuration` is different in each test function